sycl/include/sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp

```diff
@@ -29,6 +29,33 @@ namespace {
     return *this;                                                              \
   }
 
+// compare strings on compile time
+constexpr bool compareStrs(const char *Str1, const char *Str2) {
+  return std::string_view(Str1) == Str2;
+}
+
+// filter properties with AllowedPropsTuple via name checking
+template <typename TestProps, typename AllowedPropsTuple>
+struct PropertiesAreAllowed {};
+
+template <typename TestProps, typename... AllowedProps>
+struct PropertiesAreAllowed<TestProps, std::tuple<const AllowedProps...>> {
+  static constexpr const bool allowed =
+      (compareStrs(detail::PropertyMetaInfo<TestProps>::name,
+                   detail::PropertyMetaInfo<AllowedProps>::name) ||
+       ...);
+};
+
+template <typename... Ts>
+using tuple_cat_t = decltype(std::tuple_cat(std::declval<Ts>()...));
```

Comment:
Doing type-list operations with std::tuple is unnecessary slow to compile. `mp_list` is faster to compile.